#158325966 receive and implement feedback

madeofhuman opened the pull request:
#### What does this PR do?
- creates new branch to receive feedback

#### Description of Task to be completed
- create a new branch for feedback reception

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories
[#158325966](https://www.pivotaltracker.com/story/show/158325966)

#### Screenshots (if appropriate)
N/A

#### Questions
N/A

[FInishes #158325966]

madeofhuman committed:
Merge pull request #4 from madeofhuman/ft/157423408/user-dashboard-page

#157423408 Add user dashboard page

madeofhuman committed:
Add admin dashboard page

madeofhuman committed:
Merge pull request #5 from madeofhuman/ft/157424449/admin-dashboard-page

#157424449 Add admin dashboard page

madeofhuman committed:
feature: add admin/request page

madeofhuman committed:
Merge pull request #6 from madeofhuman/ft/157424622/admin-request-page

#1574246622 Add admin/request page

madeofhuman committed:
feature: add admin/resolve button` ` [Fixes: #157424622]

madeofhuman committed:
Merge pull request #7 from madeofhuman/ft/157424622/add-admin-resolve-button

[Fixes #157424622] add admin/resolve button

madeofhuman committed:
feature: add user create request page

madeofhuman committed:
Merge pull request #8 from madeofhuman/ft/157423855/user-create-request-page

#157423855 Add user/create-request page

madeofhuman committed:
bug: fix inactive links

madeofhuman committed:
Merge pull request #9 from madeofhuman/bg/157507413/inactive-link-fix

#157507413 fix inactive links

madeofhuman committed:
Bug: Fix signin and signup buttons

madeofhuman committed:
Add logout button

madeofhuman committed:
Add admin/new request page

madeofhuman committed:
Merge pull request #10 from madeofhuman/bg/157507413/login-signup-buttons-redirect

#157507413 Activate signup buttons and add admin/new page

madeofhuman committed:
Fix for Github Pages post method error

madeofhuman committed:
Merge pull request #11 from madeofhuman/bg/157518579/refactor-signin-redirect

#157518579 Fix for Github Pages post method error

madeofhuman committed:
Add Badges

madeofhuman committed:
Add legit badges

madeofhuman committed:
Merge pull request #12 from madeofhuman/ch/157499358/for-the-badge

#157499358 Add legit badges

madeofhuman committed:
Fix signin/signup redirection url typo

madeofhuman committed:
Merge pull request #14 from madeofhuman/bg/157522894/signin-url-typo

#157522894 Fix signin/signup redirection url typo

madeofhuman committed:
Add package.json and install nodemon

madeofhuman committed:
Merge pull request #16 from madeofhuman/ch/157720412/init-add-nodemon

#157720412 Add package.json and install nodemon

madeofhuman committed:
Add eslint airbnb style guide

madeofhuman committed:
Merge pull request #17 from madeofhuman/ch/157693436/add-eslint-airbnb

#157693436 Add eslint airbnb style guide

madeofhuman committed:
Add babel for transpiling ES6 to ES6

madeofhuman committed:
Add babel for transpiling ES6 code to ES5

madeofhuman committed:
Merge pull request #18 from madeofhuman/ch/157693751/babel-transpile-es6-es5

#157693751 Add babel for transpiling ES6 code to ES6

madeofhuman committed:
Revert "#157693751 Add babel for transpiling ES6 to ES6"

madeofhuman committed:
Merge pull request #19 from madeofhuman/revert-18-ch/157693751/babel-transpile-es6-es5

Revert "#157693751 Add babel for transpiling ES6 code to ES5"

madeofhuman committed:
Merge pull request #20 from madeofhuman/ch/157693751/babel-transpile-es6-es5

#157693751 Add babel for transpiling ES6 code to ES5

madeofhuman committed:
Add mocha, chai, chai-http for testing

madeofhuman committed:
Merge pull request #21 from madeofhuman/ch/157693455/mocha-chai-chaihttp-testing

#157693455 Add mocha, chai, chai-http for testing

madeofhuman committed:
Integrate TravisCI

madeofhuman committed:
Merge pull request #22 from madeofhuman/ch/157694127/travisci-integration

#157694127 Integrate TravisCI

madeofhuman committed:
Add coveralls for code coverage

madeofhuman committed:
Merge pull request #24 from madeofhuman/ch/157694144/coveralls-code-coverage

#157694144 Add coveralls for code coverage

madeofhuman committed:
Add coveralls for code coverage

madeofhuman committed:
Merge pull request #25 from madeofhuman/ch/157694144/coveralls-code-coverage

# 157694144 Add coveralls for code coverage

madeofhuman committed:
Fix TravisCI mocha permission denied

madeofhuman committed:
Fix TravisCI mocha permission denied

madeofhuman committed:
Fix TravisCI mocha permission denied

madeofhuman committed:
Fix TravisCI mocha permission denied

madeofhuman committed:
Fix TravisCI mocha permission denied

madeofhuman committed:
Fix TravisCI mocha permission denied

madeofhuman committed:
Fix TravisCI mocha permission denied

madeofhuman committed:
fixed untracked files

madeofhuman committed:
Fix TravisCI mocha permission denied

madeofhuman committed:
Fix TravisCI mocha permission denied

madeofhuman committed:
Fix TravisCI mocha permission denied

madeofhuman committed:
Fix TravisCI mocha test failure

madeofhuman committed:
Fix TravisCI mocha test failure

madeofhuman committed:
Fix TravisCI mocha test failure

madeofhuman committed:
Fix TravisCI mocha test failure

madeofhuman committed:
Fix TravisCI mocha test failure

madeofhuman committed:
Merge pull request #26 from madeofhuman/bg/157721708/mocha-permission-travisci

#157721708 Fix mocha permission denied on TravisCI

madeofhuman committed:
Add coveralls and Istanbul

madeofhuman committed:
Add nyc and remove instanbul

madeofhuman committed:
Merge pull request #27 from madeofhuman/bg/157721987/coveralls-missing-script

#157721987 Fix TravisCI coveralls missing script error

madeofhuman committed:
Add coveralls badge

madeofhuman committed:
Merge pull request #28 from madeofhuman/bg/157721987/coveralls-missing-script

[Fixes #157721987] Add coveralls badge

madeofhuman committed:
Add maintenabilty badge from codeclimate

madeofhuman committed:
Add maintenabilty badge from codeclimate

madeofhuman committed:
Merge pull request #29 from madeofhuman/ch/157722217/codeclimate-badge

#157722217 Add maintainability badge from codeclimate

madeofhuman committed:
Create /GET user/requests endpoint

madeofhuman committed:
Merge pull request #30 from madeofhuman/ft/157693416/get-users-requests

#157693416 Create /GET user/requests endpoint

madeofhuman committed:
feat(api1Test): remove call to api file from test

- remove redundnant call to api1 file

[Finishes #157693416]

madeofhuman committed:
fix(gitignore): ignore nyc and coverage folders

- ignore nyc and coverage dierctories

[Fixes: #157694144]

madeofhuman committed:
fix(package.json): add exclusion rule for nyc

- exclude UI and test directories

[Fixes: 157694144]

madeofhuman committed:
fix(gitignore): reorder dependencies

- appropriately list dependencies as production or dev

[Fixes: #157693416]

ChidinmaOrajiaku committed:
Merge pull request #31 from madeofhuman/ft/157693416/get-users-requests

#157693416 Refactor test files, gitignore, and nyc script

madeofhuman committed:
feat(route): add /POST users/requests route

- create /POST /users/requests route with tests
- add request validation module
- reorder dependency listing
- refactor database module

[Finishes: #157693893]

madeofhuman committed:
feat(route): add /GET users/request route

- create /GET users/request route with tests

[Finishes: #157693839]

madeofhuman committed:
feat(route): add /DELETE users/request/:requestId

- create a /DELETE users/request/:requestId route with tests

[Finishes: #157694151]

madeofhuman committed:
feat(route): add PUT users/requests/:requestId

- create /PUT users/requests/:requestId route with tests

[Finishes: #157694008]

madeofhuman committed:
ch(ui): refactor ui

- replace dummy text
- make forms more visible
- increase table padding
- center new request form title text
- align logout button with profile image
- increase size of request view card
- increase the size of buttons on request card
- make js ES6-compliant

[Finishes: #157507329]

madeofhuman committed:
Merge pull request #33 from madeofhuman/ft/157693839/get-users-request

#157693839 /GET users/request/:requestId

madeofhuman committed:
Merge branch 'staging-api-v1' into ft/157694151/delete-user-request

madeofhuman committed:
Merge pull request #34 from madeofhuman/ft/157694151/delete-user-request

feat(route): add /DELETE users/request/:requestId

madeofhuman committed:
Merge branch 'staging-api-v1' into ft/157694008/put-users-request

madeofhuman committed:
Merge pull request #35 from madeofhuman/ft/157694008/put-users-request

feat(route): add PUT users/requests/:requestId

madeofhuman committed:
bg(test): fix typo in deleteRequest method and test

- fix typo in database variable name

[Finishes: #157811210]

madeofhuman committed:
Merge pull request #36 from madeofhuman/ch/157507329/html-refactor

#157507329 refactor ui code

madeofhuman committed:
ch(database): add data persistence with postgres

- connect postgres for data persistence
- refactor existing routes to connect to database

[Finishes: 157841838]

madeofhuman committed:
Merge pull request #38 from madeofhuman/ch/157841838/implement-postgres-refactor

ch(database): add data persistence with postgres

madeofhuman committed:
chore(routes): implement database and jwt auth

- implement database for data persistence
- refactor existing endpoints to use database
- implement jwt authentication for route protection
- implement user login and signup

[Finishes: #157861665 #157861665 #157860962 #157860826]

madeofhuman committed:
remove createdb script in travis.yml

madeofhuman committed:
Merge pull request #39 from madeofhuman/ch/157861665/jwt-protected-routes

157861665 implement database and jwt authentication

madeofhuman committed:
feat(route): add GET /requests routes for admin

- create a GET /requests endpoint for admin to see all requests
- allow only users with admin status access the endpoint

[Finishes #157862242]

madeofhuman committed:
Merge pull request #40 from madeofhuman/ft/157862242/admin-get-all-requests

157862242 add GET /requests route for admin

madeofhuman committed:
feat(routes): add admin approve request route

- create admin-privileged route for request approval by id

[Finishes #157862335]

madeofhuman committed:
Merge pull request #41 from madeofhuman/ft/157862335/admin-approve-request

157862335 add admin approve request route

madeofhuman committed:
feat(route): add admin disapprove request route

- create admin-privileged route for request disapproval by id

[Finishes #157862407]

madeofhuman committed:
Merge pull request #42 from madeofhuman/ft/157862407/admin-disapprove-request

feat(route): add admin disapprove request route

madeofhuman committed:
feat(route): add admin resolve request route

- create admin-privileged route for request resolution by id

[Finishes #157862749]

madeofhuman committed:
Merge pull request #43 from madeofhuman/ft/157862749/admin-resolve-request

feat(route): add admin resolve request route

madeofhuman committed:
bug(requests): disallow request edit once approved

- make requests uneditable by user once approved by admin

[Finishes #157902573]

madeofhuman committed:
Merge pull request #44 from madeofhuman/bg/157902573/request-approve-edit-lock

157902573 bug(requests): disallow request edit once approved

madeofhuman committed:
chore(hosting): create configuration for heroku

- add configuration files for Heroku hosting

[Finishes #15769418]

madeofhuman committed:
Merge pull request #45 from madeofhuman/ch/157694189/heroku-config

chore(hosting): create configuration for heroku

madeofhuman committed:
Edit procfile with reltive path to index.js

madeofhuman committed:
Merge pull request #46 from madeofhuman/ch/157694189/heroku-config

Edit procfile with relative path to index.js

madeofhuman committed:
Edit procfile with reltive path to index.js

madeofhuman committed:
Merge pull request #47 from madeofhuman/ch/157694189/heroku-config

Edit procfile with reltive path to index.js

madeofhuman committed:
configure environment variable and edit start script

madeofhuman committed:
Merge pull request #48 from madeofhuman/ch/157694189/heroku-config

configure environment variable and edit start script

madeofhuman committed:
Modify procfile to run npm start

madeofhuman committed:
Merge pull request #49 from madeofhuman/ch/157694189/heroku-config

Modify procfile to run npm start

madeofhuman committed:
Modify db.js to use connection object

madeofhuman committed:
Merge pull request #50 from madeofhuman/ch/157694189/heroku-config

Modify db.js to use connection object

madeofhuman committed:
Modify db.js to use connection object

madeofhuman committed:
Merge pull request #51 from madeofhuman/ch/157694189/heroku-config

Modify db.js to use connection object

madeofhuman committed:
Modify db.js to connect to Heroku

madeofhuman committed:
Merge pull request #52 from madeofhuman/ch/157694189/heroku-config

Modify db.js to connect to Heroku

madeofhuman committed:
End connection to server after user signup

madeofhuman committed:
Merge pull request #53 from madeofhuman/ch/157694189/heroku-config

End connection to server after user signup

madeofhuman committed:
Call res.end() after successful response to client

madeofhuman committed:
Call res.end() after successful response to client

madeofhuman committed:
Merge pull request #54 from madeofhuman/ch/157694189/heroku-config

Ch/157694189/heroku config

madeofhuman committed:
Specify db server in connection string

madeofhuman committed:
Merge pull request #55 from madeofhuman/ch/157694189/heroku-config

Specify db server in connection string

madeofhuman committed:
Change to Pool connection

madeofhuman committed:
Merge pull request #56 from madeofhuman/ch/157694189/heroku-config

Change to Pool connection

madeofhuman committed:
Change to Client connection

madeofhuman committed:
Remove call to res.end()

madeofhuman committed:
Merge pull request #57 from madeofhuman/ch/157694189/heroku-config

Ch/157694189/heroku config

madeofhuman committed:
Log token to test generation on heroku

madeofhuman committed:
Merge pull request #58 from madeofhuman/ch/157694189/heroku-config

Log token to test generation on heroku

madeofhuman committed:
Log token to test generation on heroku

madeofhuman committed:
Merge pull request #59 from madeofhuman/ch/157694189/heroku-config

Log token to test generation on heroku

madeofhuman committed:
Bug(Heroku): Request timeout and empty res body

- Change to client connection
- Set token in response header instead of body

[Fixes #157694189]

madeofhuman committed:
chore(gitignore): ignore auto-generated folders

- ignore auto-generated folders

madeofhuman committed:
Merge pull request #60 from madeofhuman/ch/157694189/heroku-config

#157694189 heroku hosting configuration

madeofhuman committed:
chore(env): define NODE_ENV with cross-env

- Use cross-env to define NODE_ENV for production, dev, and test
- Configure database connection to use different DBs for each NODE_ENV
mode

[Delivers #157954446]

madeofhuman committed:
bug(RequestController): Invlidate null requests

- Prevent requests with null values from saving to db
- Create helper functions for token and user input validation
- Rename class files according to accepted convention

[Delivers #157954446]

madeofhuman committed:
chore(test): create tests with database connection

- Create tests for user signup with database connection

[Finishes #157860826]

madeofhuman committed:
chore(readme): update readme with installation instruction

- add installation intruction to readme
- update seed.sql

[Finishes #157981510]

madeofhuman committed:
Merge pull request #61 from madeofhuman/157954446-bug-valdiate-request-body

#157954446 Validate user request input

madeofhuman committed:
Update Readme.md

Fix typo and formatting

madeofhuman committed:
bug(filename): rename models and controllers

- rename model and controller files in line with convention

[Finishes #157983331]

madeofhuman committed:
Merge pull request #62 from madeofhuman/157954446-bug-valdiate-request-body

#157983331 rename models and controllers

madeofhuman committed:
chore(RequestController): abstract more functions

- Abstract more repeated functions into helper methods

[Delivers #157901375]

madeofhuman committed:
chore(tests): write more tests

- cover more edge cases with more tests

[Delivers #157992012]

madeofhuman committed:
Merge pull request #63 from madeofhuman/157954446-bug-valdiate-request-body

#157954446 abstract more functions and add more tests

madeofhuman committed:
chore(readme): make readme more descriptive

- make readme file more descriptive for collaborators

[Finishes #157981510]

madeofhuman committed:
chore(readme): modify readme

- modify error in table
[Fixes #157981510]

madeofhuman committed:
chore(readme): modify error in readme

- fix typo in app links
[Fixes #157981510]

madeofhuman committed:
bug(readme): fix code markdown formatting

- fix the formatting for code markdown

[Fixes #157981510]

madeofhuman committed:
Merge pull request #64 from madeofhuman/157981510-chore-update-readme

#157981510 chore update readme

madeofhuman committed:
bug(readme): fix type in readme

-fix typo in readme
[Finishes #157981510]

madeofhuman committed:
chore(api): implement feedback from lfa

- astract createUser function
- prevent hashed password from being returned to the user
- make duplicate email error message user-friendly
- comment code

[Delivers #157996170]

madeofhuman committed:
Merge pull request #65 from madeofhuman/157996170-chore-apply-lfa-feedback

#157996170 implement feedback from lfa

madeofhuman committed:
chore(feedback): implement lfa feedback

- remove commented code
- write functions in es6 format

[Delivers #157996170]

madeofhuman committed:
Merge pull request #66 from madeofhuman/157996170-chore-apply-lfa-feedback

#157996170: implement lfa feedback

madeofhuman committed:
chore(feedback): implement lfa feedback

- change no-request-found status to 200, and 'error' key to 'message'
- set request status in model instead of hard-coding in controller
- trim login email
- change duplicate-email message status code to 409
- change no-email-found-on-login error status to 403
- change admin authentication failure status to 401
- remove returned id in request creation confirmation
- set authorization bearer token in header

[Delivers #157996170]

madeofhuman committed:
bug(readme): fix typo in readme

- fix markdown typo

[Fixes #157981510]

madeofhuman committed:
chore(tests): add more tests

- add more tests to catch more edge cases

[Delivers #157996170]

madeofhuman committed:
bug(test): udpate access token in test file

- update access token in test file

[Fixes #157996170]

madeofhuman committed:
chore(test): add more tests

- implement more tests to catch more edge cases

[Delivers #157996170]

madeofhuman committed:
chore(messages): make error messages user-friendly

- make error messages user-friendly

[Delivers #157996170]

madeofhuman committed:
bug(feedback): validate user login input

- validate email in user login
- remove trim function on user account creation input

[Delivers #157996170]

madeofhuman committed:
bug(test): fix failing request update test

- fix failing request update test due to non-similar data in local db
and ElephantSQL db

[Fixes #157996170]

madeofhuman committed:
Merge pull request #67 from madeofhuman/157996170-chore-apply-lfa-feedback

#157996170 apply LFA feedback

madeofhuman committed:
bug(UI): implement lfa feedback on UI and refactor

- use checkbox or dropdown to set request type
- seperate approve and view buttons
- remove status column in admin table

[Finishes #157996170]

madeofhuman committed:
Merge pull request #68 from madeofhuman/157996170-implment-lfa-feedback

#157996170 implement LFA's feedback on UI

madeofhuman committed:
chore(git): remove auto-generated folders

- remove .vscode folder

[Finishes #157996170]

madeofhuman committed:
bug(requestController): remove created_at return

- remove created_at field return after request creation

[Finishes #158018354]

madeofhuman committed:
bug(requestController): validate requestId first

- validates requestId first to check if type is valid number

[Finishes #158018354]

madeofhuman committed:
bug(tests): fix failing test

- fix failing tests

[Fixes: #158018354]

madeofhuman committed:
bug(request): fix unnecessary data return

- remove created_at and updated_at fields when returning data to the
user after a request update

[Finishes #158018354]

madeofhuman committed:
Merge pull request #70 from madeofhuman/158018354-bug-fix-unnecesssary-data-return

#158018354 remove unnecessary data return and validates url parameters

madeofhuman committed:
bug(validation): implement feedback on validations

- make error messages for login and signup more descriptive
- ensure response message consistency across the application
- return jwt token in response body
- validate user input for signup and login
- disallow special characters in user input for signup and login
- properly validate user email
- approve, disapprove or resolve requests based on state
- write more tests to cover more edge cases

[Delivers ##3985384]

madeofhuman committed:
bug(TravisCI): fix failing delete and update tests

- reference requestId on elephantsql

[Fixes ##3985384]

madeofhuman committed:
feat(UI): add 'back to dashboard' button

- add button to enable users return to the dashboard from other pages

[Finishes #158027824]

madeofhuman committed:
Merge pull request #71 from madeofhuman/158027824-chore-back-to-dashbord-button

158027824 add 'back to dashboard' button

madeofhuman committed:
bug(UI): fix back button alignment

- fix alignment for back button

[Fixes #158027824]

madeofhuman committed:
bug(tests): fix failing update and delete tests

- get last request id from db instead of hard-coding an id

[Fixes #158028459]

madeofhuman committed:
bug(tests): fix failing update and delete tests

- get last request id from db instead of hard-coding an id

[Fixes #158028459]

madeofhuman committed:
bug(test): refactor tests

- refactor tests

[Fixes #158028459]

madeofhuman committed:
Merge pull request #73 from madeofhuman/158028459-chore-descriptive-error-messages

3985384 apply validation feedback

madeofhuman committed:
Merge branch 'staging-api-v1' into staging

madeofhuman committed:
chore(gitignore): ignore log file

- add log file to gitignore

[Delivers #158131872]

madeofhuman committed:
chore(UI): reorganise directory structure

- reorganise UI directory structure

[Delivers #3992197]

madeofhuman committed:
chore(eslint): add browser mode to eslint config

- add browser rule to eslint config file

[Delivers #3992197]

madeofhuman committed:
feat(route): declare routes for client

- add client routes

[Delivers #3992197]

madeofhuman committed:
chore(ui): reorganise file directory

- track assets directory that was re-organised

[Delivers #3992197]

madeofhuman committed:
chore(ui): reorganise ui directory

- track assets folder that was reorganised

[Delivers #3992197]

madeofhuman committed:
feat(client-auth): implement signup and login

- implement user signup and login on client
- notify user of incorrect credentials

[Delivers #158131872]
[Delivers #158132187]
[Delivers #158132407]
[Delivers #157437963]

madeofhuman committed:
chore(auth): protect client routes

- add protection for priviledged routes on the client

[Delivers #158228967]

madeofhuman committed:
feat(client): implement user dashboard

- connect user dashboard with users/requests/ api endpoint for viewing
all requests

[Delivers ##158132449]

madeofhuman committed:
feat(client): add logout functionality

- add logout functionality for logged-in users

[Delivers #158177575]

madeofhuman committed:
feat(client): implement user-view-request

- connect view request client page to users/requests/:requestId api
endpoint

[Delivers #158176292]

madeofhuman committed:
feat(client): create 404 page

- create 404 page for invalid route navigation on client

[Delivers #158178744]

madeofhuman committed:
feat(client): implement user-create-request

- connect new request page on client to POST users/requests api endpoint

[Delivers #158132481]

madeofhuman committed:
feat(client): implement user-edit-request

- connect edit request client page to PUT users/request/:requestId api
endpoint

[Delivers #158176350]

madeofhuman committed:
feat(client): implement user-delete-request

- connect delete request button on client/request page to DELETE
/users/requests:requestId api endpoint

[Delivers #158176396]

madeofhuman committed:
feat(client): implement admin-view-requests

- connect admin dashboard on client to GET /requests api endpoint to display
all requests in the system

[Delivers #158193676]

madeofhuman committed:
feat(client): implement admin-view-request

- connect admin/view client page to GET /requests/:requestId api
endpoint to display the details of a request

[Delivers #158213946]

madeofhuman committed:
feat(client): implement admin request actions

- connect request action buttons on admin/view page to their respective
api endpoints (approve, disapprove, resolve)

[Delivers #158213971]

madeofhuman committed:
chore(validation): fix verbose error messages

- make some verbose validation error messages more user-friendly

[Delivers #158216135]

madeofhuman committed:
bug(test): fix failing tests

- fix failing tests due to edits to route and valdation files

[Delivers #158229568]

madeofhuman committed:
Merge pull request #74 from madeofhuman/3992197-client-api-connect

#3992197 implement front-end app

madeofhuman committed:
bug(client): fix typo in edit request form

- fix typo in edit request form

[Fixes ##3992197]

madeofhuman committed:
bug(client): fix typo in request type on create

- fix spelling error in request type in request creation form that
prevents maintenance requests from being created

[Fixes #158132481]

madeofhuman committed:
bug(typo): fix typo in request edit form

- fix typo in request type in request edit form that prevents type from
being set to 'maintenance'

[Fixes #158176350]

madeofhuman committed:
Merge pull request #75 from madeofhuman/3992197-client-api-connect

3992197 correct request form typo

madeofhuman committed:
chore(docs): add api documentation with Swagger

- add api documentation with Swagger

[Finishes #158021032]

madeofhuman committed:
bug(docs): fix Swagger not working on Heroku

- fix for Swagger docs crashing app on Heroku

[Fixes #158021032]

madeofhuman committed:
chore(Heroku): remove unnecessary Procfile

- removes unnecessary Procfile

[Fixes #158021032]

madeofhuman committed:
bug(docs): fix for Swagger crashing app on Heroku

- fixes Swagger docs crashing app on Heroku

[Fixes #158021032]

madeofhuman committed:
Merge pull request #76 from madeofhuman/158021032-chore-swagger-api-docs

#158021032 add api documentation with Swagger

madeofhuman committed:
chore(env): create sample env file

- create env template file for creating .env

[Finishes #158398788]

madeofhuman committed:
Merge pull request #79 from madeofhuman/158398788-chore-create-sample-env

#158398788 create sample env file

madeofhuman committed:
chore(client): edit 404 message

- replace the emoticon with an emoji

[Delivers #158214663]

madeofhuman committed:
chore(route): separate user & admin client routes

- put user and admin client routes in separate files

[Delivers #158214663]

madeofhuman committed:
chore(client): abstract duplicated code into function

- create function for form submit

[Delivers #158214663]

madeofhuman committed:
chore(client): fix duplicated code

- abstract duplicated code into modular functions

[Delivers #158214663]

humanityjs commented on .babelrc:
```diff
@@ -0,0 +1,4 @@
+{
+  "presets": ["es2015", "stage-2"],
+  "plugins": []
+}
```
New space required

humanityjs commented on .eslintrc:
```diff
@@ -0,0 +1,12 @@
+    "no-console": "off",
+    "import/prefer-default-export": "off"
+  }
+}
```
New space required

humanityjs commented on .gitignore:
```diff
@@ -0,0 +1,17 @@
+
+.env
+
+server/access.log
```
New space required

humanityjs commented on .travis.yml:
```diff
@@ -0,0 +1,12 @@
+before_script:
+ - chmod 777 ./node_modules/.bin/mocha
+
+after_success: npm run coverage
```
New space required

humanityjs commented on README.md:
```diff
@@ -1,11 +1,85 @@
+  3. PostgreSQL
+
+## 2. Coding style
+  Airbnb style
```
Can you add the link to the style guide?

humanityjs commented on UI/assets/js/adminDashboard.js:
```diff
@@ -0,0 +1,75 @@
+
+/* Set welcome message */
+const user = JSON.parse(window.localStorage.getItem('User'));
+console.log(user.firstName);
```
Please remove the console

humanityjs commented on UI/assets/js/adminDashboard.js:
```diff
@@ -0,0 +1,75 @@
+const sortGroup = document.getElementById('sort-group');
+
+/* Set welcome message */
+const user = JSON.parse(window.localStorage.getItem('User'));
```
I'll suggest you use a more descriptive name for your user object. Instead of saving it as `User`, you might want to save it as `[your site]User` or something.

humanityjs commented on UI/assets/js/adminDashboard.js:
```diff
@@ -0,0 +1,75 @@
+/* Set welcome message */
+const user = JSON.parse(window.localStorage.getItem('User'));
+console.log(user.firstName);
+username.innerHTML += `${user.firstName}.`;
```
Here, you are assuming that the user object will be found in the local storage. What happens when the user is not found?

madeofhuman replied:
With the way I wrote it, the User object has to be thee because it's set after login. The user can't access the page otherwise. But I see your point, I'll add a check for that.

humanityjs commented on UI/assets/js/adminDashboard.js:
```diff
@@ -0,0 +1,75 @@
+
+fetch(requestsUrl, {
+  headers: {
+    Authorization: window.localStorage.getItem('Authorization'),
```
You can use a more descriptive name for your auth token as well so as to distinguish it from other tokens that might be present with the same name

humanityjs replied:
ALso, this is assuming that you have an auth token in your localstorage. What happens when there's no token in the localstorage? Since this page is supposed to be an auth page, I'll even suggest that you don't even make the request if there's no token.

humanityjs commented on UI/assets/js/adminDashboard.js:
```diff
@@ -0,0 +1,75 @@
+    Authorization: window.localStorage.getItem('Authorization'),
+  },
+})
+  .then(response => response.json())
```
Is there any reason why you have to return `response.json`?

humanityjs commented on UI/assets/js/adminDashboard.js:
```diff
@@ -0,0 +1,75 @@
+        </table>
+      `;
+      const requestsTable = document.getElementById('requests-table');
+      data.result.forEach((request) => {
```
Just curious, is there a better way of doing this?

humanityjs commented on UI/assets/js/adminRequestOperations.js:
```diff
@@ -0,0 +1,105 @@
+        window.location.replace(`/admin/view?id=${param}`);
+      }
+      output.innerHTML = 'request disapproved';
+      const redirectDashboard = () => window.location.replace('/admin');
```
Why do you choose `.replace` instead of `.redirect` or `.href`?

humanityjs commented on UI/assets/js/adminView.js:
```diff
@@ -0,0 +1,81 @@
+})
+  .then(response => response.json())
+  .then((data) => {
+    console.log(data);
```
Please remove this console

humanityjs commented on UI/assets/js/adminView.js:
```diff
@@ -0,0 +1,81 @@
+      output.innerHTML = 'The request id you entered does not exist';
+      window.location.replace('/admin');
+    } else {
+      wrapper.innerHTML = `
```
So, instead of having to first create the html and then add the data later, I'll suggest that you create all the variables you'll need to populate the html and then assign it to them directly since you are already using a template string.

humanityjs commented on UI/views/404.html:
```diff
@@ -0,0 +1,48 @@
+
+  <script>
+    const backButton = document.getElementById('back-btn');
+    backButton.addEventListener('click', () => window.history.back());
```
Nice thinking here 👍

humanityjs commented on package.json:
```diff
@@ -0,0 +1,70 @@
+  },
+  "homepage": "https://github.com/madeofhuman/Maintenance-Tracker-App#readme",
+  "devDependencies": {
+    "coffeescript": "^2.3.0",
```
What are you using CoffeeScript for?

humanityjs commented on server/controllers/RequestController.js:
```diff
@@ -0,0 +1,351 @@
+    )
+      .then((result) => {
+        if (result.rows < 1) {
+          return res.status(200).json({
```
Why are you returning status 200 and sending statusCode 500 in the response body?

humanityjs commented on server/controllers/RequestController.js:
```diff
@@ -0,0 +1,351 @@
+            statusCode: 500,
+            error: [],
+            message: 'You have no request with that id, please try another request id',
+            request: result.rows,
```
Since you are assuming that the request has failed and there's no request with that id, why returning `request.rows`?

humanityjs commented on server/controllers/RequestController.js:
```diff
@@ -0,0 +1,351 @@
+          return res.status(200).json({
+            statusCode: 200,
+            error: [],
+            message: 'There are no requests in the system. Do we even have people using this app?',
```
Yo! Nice work with your response messages. They are real nice

humanityjs replied:
But, as much as I love your response messages, they can sometimes be distracting. So I'll suggest you make the message more professional since you are not building a social app.

humanityjs commented on server/controllers/RequestController.js:
```diff
@@ -0,0 +1,351 @@
+      .then((result) => {
+        if (result.rows < 1) {
+          return res.status(200).json({
+            statusCode: 200,
```
Why is this showing a 200 status code while the message is saying something totally different?

humanityjs commented on server/controllers/UserController.js:
```diff
@@ -0,0 +1,106 @@
+          return res.status(500).json({
+            statusCode: 500,
+            error: 'Internal Server Error',
+            message: 'Oops! Another bug must have crawled into our systems, but we are right on it! Please try your request later :\'(',
```
And since you are using the same message in multiple places, you can abstract this and create an helper function for it.

humanityjs commented on server/controllers/UserController.js:
```diff
@@ -0,0 +1,106 @@
+          return res.status(403).json({
+            statusCode: 403,
+            error: 'Forbidden',
+            message: 'The email address you entered does not exist',
```
I'll suggest that instead of telling the user that the email does not exist, you should just tell the user that his authentication failed.

humanityjs commented on server/controllers/UserController.js:
```diff
@@ -0,0 +1,106 @@
+          });
+        }
+
+        const payload = {
```
You should only prepare your payload after you have verified the password.

humanityjs reviewed (commented):
Nice job so far. Please take note of the comments and address them as necessary.

madeofhuman committed:
Merge pull request #80 from madeofhuman/158214663-chore-refactor-client

#158214663 refactor client code

madeofhuman committed:
chore(feedback): remove coffeescript package

- remove unnecessary coffeescript package

[Delivers #158478970]

madeofhuman committed:
chore(feedback): add trailing spaces

- add trailing spaces on last lines of code

[Delivers #158478970]

madeofhuman committed:
chore(feedback): add link to style guide

- add link to eslint airbnb style guide in README

[Delivers #158478970]

madeofhuman committed:
chore(feedback): remove console.log

- remove console.log calls and replace error logging with console.error

[Delivers #158478970]

madeofhuman committed:
chore(feedback): edit request api res.statusCode

- changes getRequest API statusCode to 200

[Delivers #158470]

madeofhuman committed:
chore(feedback): modify userLogin process

- define payload only after user credentials have been successfully
  authentnicated

  [Delivers #158478970]

madeofhuman committed:
chore(feedback): modify userLogin auth message

- return less-specific authentication error message for security reasons

[Delivers #158478970]

madeofhuman committed:
chore(api): abstract api responses

- abstract repeated api responses into helper functions

[Delivers #158478970]

madeofhuman committed:
bug(api): fix admin-get-request error

- add omitted parameter in api response method

[Fixes #158478970]

madeofhuman committed:
bug(api): fix user-update-request error

- supply omitted api message function parameter

[Fixes #158478970]

madeofhuman committed:
chore(localstorage): rename auth and user objects

- use more and unique names for auth and user objects in localstorage

[Delivers #158478970]

madeofhuman committed:
chore(client): refactor request table

- replace forEach loop with for loop

[Delivers #158478970]

madeofhuman committed:
bug(test): fix failing test

- fix failing tests resulting from changing some api response messages

[Fixes #158478970]

madeofhuman committed:
Merge pull request #81 from madeofhuman/158478970-chore-implement-cr-feedback

#158478970 implement code review feedback

madeofhuman committed:
bug(auth): fix redirect loop when no auth

- fix redirect loop when no auth resulting from expired token

[Fixes #158478970]

madeofhuman committed:
chore(migration): streamline database creation

- streamline database creation process with a script that creates dev and
  test databases and their tables

  [Delivers #158555431]

madeofhuman committed:
Merge pull request #82 from madeofhuman/158555431-chore-streamline-db-creation

#158555431 streamline db migraton

madeofhuman committed:
chore(npm): update bcrypt and untrack package-lock.json
- update bcrypt to fix vulnerability warning
- untrack package-lock.json

madeofhuman committed:
add cors and return user info on login

madeofhuman committed:
return updated request after admin action

madeofhuman committed:
fix create request validation that accepts empty string as request item and description value

madeofhuman committed:
feat(signup and api responses): login user on signup and sanitize error messages
- login user on signup
- sanitize error messages
